examples/analysis/IDE.flix

```diff
@@ -1,182 +1,210 @@
-namespace IDE {
+enum Value {
+    case Top,
+    case Cst(Int),
+    case Bot
+}
 
-    enum Value {
-        case Top,
-        case Cst(Int),
-        case Bot
-    }
+def sum(e1: Value, e2: Int): Value = match e1 {
+    case Value.Cst(n) => Value.Cst(n+e2)
+    case _ => e1
+}
 
-    def sum(e1: Value, e2: Int): Value = match e1 {
-        case Value.Cst(n) => Value.Cst(n+e2)
-        case _ => e1
-    }
+def prod(e1: Value, e2: Int): Value = match (e1, e2) {
+    case (Value.Cst(n), _) => Value.Cst(n*e2)
+    case (Value.Bot, _) => Value.Bot
+    case (_, zero) => if(zero == 0) Value.Cst(0) else e1
+    case _ => e1
+}
 
-    def prod(e1: Value, e2: Int): Value = match (e1, e2) {
-        case (Value.Cst(n), _) => Value.Cst(n*e2)
-        case (Value.Bot, _) => Value.Bot
-        case (_, zero) => if(zero == 0) Value.Cst(0) else e1
-        case _ => e1
-    }
+instance LowerBound[Value] {
+    pub def minValue(): Value = Value.Bot
+}
 
-    instance LowerBound[Value] {
-        pub def minValue(): Value = Value.Bot
+instance Eq[Value] {
+    pub def eq(x: Value, y: Value): Bool = match (x, y) {
+        case (Top, Top) => true
+        case (Cst(a), Cst(b)) => a == b
+        case (Bot, Bot) => true
+        case _ => false
     }
+}
 
-    instance Eq[Value] {
-        pub def eq(x: Value, y: Value): Bool = match (x, y) {
-            case (Top, Top) => true
-            case (Cst(a), Cst(b)) => a == b
-            case (Bot, Bot) => true
-            case _ => false
-        }
+instance PartialOrder[Value] {
+    pub def partialCompare(e1: Value, e2: Value): Bool = match (e1, e2) {
+        case (Value.Bot, _) => true
+        case (Value.Cst(n1), Value.Cst(n2)) => n1 == n2
+        case (_, Value.Top) => true
+        case _ => false
```

Comment:
The lack of ordering between Cst values

Reply:
What do you mean?

Reply:
Just an addition to the other comment, since I couldn't mark to different places in the code.